app/controllers/api/functions.php

```diff
@@ -1569,6 +1573,14 @@
             }
         }
 
+        $jwtExpiry = $function->getAttribute('timeout', 900);
+        $jwtObj = new JWT(App::getEnv('_APP_OPENSSL_KEY_V1'), 'HS256', $jwtExpiry, 10);
+        $apiToken = $jwtObj->encode([
+            'projectId' => $project->getId(),
+            'scopes' => $function->getAttribute('apiTokenScopes', [])
+        ]);
+
+        $headers['x-appwrite-api-token'] = $apiToken;
```

Comment:
api-token vs api-key is extremely confusing. This is exactly how we create complexity where two parallel terms split the user journey in two different paths and they are forced to memorize an Appwrite specific patch. 

I understand we can't create a regular API key for every function as this will create a huge clutter, but I'm curious by the idea of using the same `x-appwrite-key` header and SDK method with some prefix (so we can parse it internally and validate it differently) and let it behave exactly the same for the user.

Reply:
What about using Authentication header? 
`Authorization: <auth-scheme> <authorization-parameters>`

Reply:
This could be nice as we move down the line, my biggest worry is to have to explain 2 very similar methods, I would love us to abstract that complexity from developers using the API or SDK with the `setKey()` method.

Reply:
Okay, Ill use setKey and `x-appwrite-key` for both.
Would it be fine to have a bit of magic for recognizing if it's regular or dynamic key? My initial idea is to first check against API keys on project (we have project document already, so it's not network request). If we don't find it that way, then we try to parse it as JWT. If both fails, then we consider we don't have any key.

Reply:
I think performance wise it might be better if we add some prefix or other metadata behind the base64 encoding so we can parse the key in the correct way faster.